### PR TITLE
SDIT-1096 Non-association Reconciliation process

### PR DESCRIPTION
**Pull request by steverendell**
Complete range of chacks on each NA and detect extra dps NAs

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt [hunk 1]
```diff
@@ -76,7 +79,16 @@ class NonAssociationsReconciliationService(
...
+                it.restrictionType.name,
+                it.whenCreated,
+                "",
+                it.isClosed,
+                it.firstPrisonerRole.name,
+                it.secondPrisonerRole.name,
+                it.reason.name,
+                it.comment,
+              ),
             )
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt:92]: This just dumps out all the 'extra' DPS NAs

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsResourceIntTest.kt [hunk 2]
```diff
@@ -139,17 +156,18 @@ class NonAssociationsResourceIntTest : IntegrationTestBase() {
...
+      nomisApi.stubGetNonAssociationsPage(1, 10, nonAssociationsNomisPagedResponse(totalElements = numberOfNonAssociations, numberOfElements = 10, pageNumber = 1, pageSize = 10))
+      nomisApi.stubGetNonAssociationsPage(2, 10, nonAssociationsNomisPagedResponse(totalElements = numberOfNonAssociations, numberOfElements = 10, pageNumber = 2, pageSize = 10))
+      nomisApi.stubGetNonAssociationsPage(3, 10, nonAssociationsNomisPagedResponse(totalElements = numberOfNonAssociations, numberOfElements = 4, pageNumber = 3, pageSize = 10))
       (1..numberOfNonAssociations).forEach {
         val offenderNo1 = index1ToNomsId(it)
         val offenderNo2 = index2ToNomsId(it)
         nomisApi.stubGetNonAssociationsAll(offenderNo1, offenderNo2, nonAssociationNomisResponse(offenderNo1, offenderNo2))
         nonAssociationsApiServer.stubGetNonAssociationsBetween(offenderNo1, offenderNo2, nonAssociationApiResponse(if (it.toInt() % 10 == 0) "WING" else "LANDING")) // // every 10th prisoner has an WING type
       }
+      nonAssociationsApiServer.stubGetNonAssociationsPage(0, 1, nonAssociationsDpsPagedResponse(34))
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsResourceIntTest.kt:170]: For now Im stubbing this to not find any extra DPS NAs

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/NomisApiMockServer.kt [hunk 3]
```diff
@@ -1059,75 +1059,6 @@ class NomisApiMockServer : WireMockServer(WIREMOCK_PORT) {
     )
   }
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/NomisApiMockServer.kt:1061]: Just moved these funs into the right section

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**